### PR TITLE
MoveIt Task Constructor Tutorial

### PR DESCRIPTION
**Pull request by stephanie-eng**
### Description

Adds a tutorial that explains how to create a package that uses MoveIt Task Constructor to plan a basic pick and place task using the panda robot.

### TODO
- [x] Merge the mtc_pick_and_place_tutorial (containing the code example) with the pick_and_place_with_moveit_task_constructor folder (containing the documentation)
- [x] Update the code blocks in the documentation to match with the code example
- [x] Finish adding stages to the tutorial code example
- [x] Fill out some text outlining how to add the pick and place stages, and explain the individual stages
- [x] Fill out Visualization with RViz section
- [x] Add images
- [x] General cleanup, spellcheck, etc.
- [x] Fix links
- [x] Fix any build issues
- [x] Add links to tutorial text referencing MTC as well as the example code
- [x] Add instructions to use the launchfile from the code example
- [x] Put in knowledge prerequisites at the top of the tutorial

### CODE REVIEW

**Comment by stephanie-eng** [comment]
@schornakj I can't add you as a reviewer for some reason, but if you can, I'd appreciate your review since you helped me with this!

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 1]
```diff
@@ -0,0 +1,883 @@
...
+    const auto& hand_frame = "panda_hand";
+
+    // Set task properties
+    task.setProperty("group", arm_group_name);
+    task.setProperty("eef", eef_name);
+    task.setProperty("hand", hand_group_name);
+    task.setProperty("hand_grasping_frame", hand_frame);
+    task.setProperty("ik_frame", hand_frame);
+
+    moveit::task_constructor::Stage* current_state_ptr = nullptr;  // Forward current_state on to grasp pose generator
```
Comment (henningkayser) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:220]: Some of these lines are too long. I guess you could copy a clang-formatted version, possibly with a shorter line length.
Also, I wouldn't be against doing something like `namespace mtc = moveit::task_constructor`

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 2]
```diff
@@ -0,0 +1,883 @@
...
+
+Code Breakdown
+^^^^^^^^^^^^^^
+
+The top of the code includes the ROS and MoveIt Libraries that this package uses.
+
+ * ``rclcpp/rclcpp.hpp`` includes core ROS2 functionality
+ * ``moveit/planning_scene/planning_scene.h`` and ``moveit/planning_scene_interface/planning_scene_interface.h`` includes functionality to interface with the robot model and collision objects
+ * ``moveit/task_constructor/task.h``, ``moveit/task_constructor/solvers.h``, and ``moveit/task_constructor/stages.h`` include different components of MoveIt Task Constructor that are used in the example
+ * ``tf2_geometry_msgs/tf2_geometry_msgs.hpp`` and ``tf2_eigen/tf2_eigen.hpp`` won't be used in this initial example, but when will be used for pose generation when we add more stages to the MoveIt Task Constructor task.
```
Comment (henningkayser) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:273]:
```suggestion
 * ``tf2_geometry_msgs/tf2_geometry_msgs.hpp`` and ``tf2_eigen/tf2_eigen.hpp`` won't be used in this initial example, but they will be used for pose generation when we add more stages to the MoveIt Task Constructor task.
```

Maybe consider adding the includes together with the other stages

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 3]
```diff
@@ -0,0 +1,883 @@
...
+.. code-block:: c++
+
+    moveit::task_constructor::Stage* current_state_ptr = nullptr;  // Forward current_state on to grasp pose generator
+    auto stage_state_current = std::make_unique<moveit::task_constructor::stages::CurrentState>("current");
+    current_state_ptr = stage_state_current.get();
+    task.add(std::move(stage_state_current));
+
+In order to plan any robot motions, we need to specify a solver. MoveIt Task Constructor has three options for solvers:
+
+ * ``PipelinePlanner`` uses the default MoveIt planner, which is typically OMPL.
```
Comment (henningkayser) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:432]:
```suggestion
 * ``PipelinePlanner`` uses MoveIt's planning pipeline, which typically defaults to OMPL.
```

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 4]
```diff
@@ -0,0 +1,883 @@
...
+ * ``CartesianPath`` is used to move the end effector in a straight line in Cartesian space.
+ * ``JointInterpolation`` is a simple planner that interpolates between the start and goal joint states. It is typically used for simple motions as it is fast to plan, but may fail for complex motions.
```
Comment (henningkayser) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:434]:
```suggestion
 * ``JointInterpolation`` is a simple planner that interpolates between the start and goal joint states. It is typically used for simple motions as it computes pretty fast, but it doesn't support complex motions.
```

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 5]
```diff
@@ -0,0 +1,883 @@
...
+
+.. code-block:: c++
+
+    auto stage_open_hand = std::make_unique<moveit::task_constructor::stages::MoveTo>("open hand", interpolation_planner);
+    stage_open_hand->setGroup(hand_group_name);
+    stage_open_hand->setGoal("open");
+    task.add(std::move(stage_open_hand));
+
+    return task;
+    }
```
Comment (henningkayser) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:458]:
```suggestion
```

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 6]
```diff
@@ -0,0 +1,883 @@
...
+
+Running the Demo
+----------------
+
+Launch files
+^^^^^^^^^^^^
+
+We will need a launch file to launch ``move_group``, ``ros2_control``, ``static_tf``, ``robot_state_publisher``, and ``rviz``. :codedir:`Here <tutorials/pick_and_place_with_moveit_task_constructor/launch/pick_place_demo.launch.py>` is the launch file we use in the tutorials package. Put this in the launch directory of your package.
+
+To run the MoveIt Task Constructor node, we need a second launchfile to start the ``mtc_tutorial`` executable with the proper parameters. Either load your URDF, SRDF, and OMPL parameters, or use MoveIt Configs Utils to do so. Your launch file should look something like this:
```
Comment (henningkayser) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:498]:
```suggestion
To run the MoveIt Task Constructor node, we need a second launch file to start the ``mtc_tutorial`` executable with the proper parameters. Either load your URDF, SRDF, and OMPL parameters, or use MoveIt Configs Utils to do so. Your launch file should look something like this:
```

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 7]
```diff
@@ -0,0 +1,883 @@
...
+
+In order to see your robot and the MoveIt Task Constructor solutions in RViz, we'll have to make some changes to the RViz configuration. First, start RViz. The following steps will cover how to set up RViz for MoveIt Task Constructor solution visualization.
+
+1. If the **MotionPlanning** display is active, uncheck it to hide it for now.
+2. Under **Global Options**, change the **Fixed Frame** from ``map`` to ``panda_link0`` if not already done.
+3. On the bottom left of the window, click the **Add** button.
+4. Under ``moveit_task_constructor_visualization`` select **Motion Planning Tasks** and click OK. The **Motion Planning Tasks** display should appear on the bottom left.
+5. In the **Displays**, under **Motion Planning Tasks**,  change **Task Solution Topic** to ``/solution``
+
+You should see the panda arm in the main view with Motion Planning Tasks display open in the bottom left and nothing in it. Your MTC task will show up in this pane once you launch the ``mtc_tutorial`` node. If you're using ``mtc_demo.launch.py`` from the tutorials, jump back in here.
```
Comment (henningkayser) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:544]:
```suggestion
You should see the panda arm in the main view with Motion Planning Tasks display open in the bottom left and nothing in it. Your MTC task will show up in this panel once you launch the ``mtc_tutorial`` node. If you're using ``mtc_demo.launch.py`` from the tutorials, jump back in here.
```

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 8]
```diff
@@ -0,0 +1,883 @@
...
+  moveit::task_constructor::Stage* attach_object_stage =
+      nullptr;  // Forward attach_object_stage to place pose generator
+
+This next block of code creates a ``SerialContainer``. This is a container that can be added to our task and can hold several substages. In this case, we create a serial container that will contain the stages relevant to the picking action. Instead of adding the stages to the task, we will add the relevant stages to the serial container.
+
+.. code-block:: c++
+
+  {
+    auto grasp = std::make_unique<moveit::task_constructor::SerialContainer>("pick object");
+    task.properties().exposeTo(grasp->properties(), { "eef", "hand", "group", "ik_frame" });
```
Comment (henningkayser) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:608]: You could mention here that `exposeTo` declares the task properties from the parent task in the new grasp container, `configureInitFrom()` initializes them. This is necessary for allowing the contained stages to access them

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 9]
```diff
@@ -0,0 +1,883 @@
...
+.. code-block:: c++
+
+      Eigen::Isometry3d grasp_frame_transform;
+      Eigen::Quaterniond q = Eigen::AngleAxisd(M_PI / 2, Eigen::Vector3d::UnitX()) *
+                             Eigen::AngleAxisd(M_PI / 2, Eigen::Vector3d::UnitY()) *
+                             Eigen::AngleAxisd(M_PI / 2, Eigen::Vector3d::UnitZ());
+      grasp_frame_transform.linear() = q.matrix();
+      grasp_frame_transform.translation().z() = 0.1;
+
+Now, we create the ``ComputeIK`` stage, and give it the name ``generate pose IK`` as well as the ``generate grasp pose`` stage defined above. Some robots have multiple inverse kinematics solutions for a given pose - we set the limit on the amount of solutions to solve for to 8. We also set the minimum solution distance, which is a threshold on how different solutions must be: if the joint positions in a solution are too similar to a previous solution, it will be marked as invalid. Next, we configure some additional properties, and add the ``ComputeIK`` stage to the serial container.
```
Comment (henningkayser) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:657]:
```suggestion
Now, we create the ``ComputeIK`` stage, and give it the name ``generate pose IK`` as well as the ``generate grasp pose`` stage defined above. Some robots have multiple inverse kinematics solutions for a given pose - we set the limit on the amount of solutions to solve for up to 8. We also set the minimum solution distance, which is a threshold on how different solutions must be: if the joint positions in a solution are too similar to a previous solution, it will be marked as invalid. Next, we configure some additional properties, and add the ``ComputeIK`` stage to the serial container.
```

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 10]
```diff
@@ -0,0 +1,883 @@
...
+
+.. code-block:: c++
+
+  {
+    auto place = std::make_unique<moveit::task_constructor::SerialContainer>("place object");
+    task.properties().exposeTo(place->properties(), { "eef", "hand", "group", "ik_frame" });
+    place->properties().configureInitFrom(moveit::task_constructor::Stage::PARENT,
+                                          { "eef", "hand", "group", "ik_frame" });
+
+This next stage generates the poses used to place the object and compute the inverse kinematics for those poses - it is somewhat similar to the ``generate grasp pose`` stage from the pick serial container. We start by creating a stage to generate the poses and configuring some properties. We specify the pose where we want to place the object with a ``PoseStamped`` message from ``geometry_msgs`` - in this case, we choose ``y = 0.5``. We then pass the target pose to the stage with ``setPose``.  Next, we use ``setMonitoredStage`` and pass it the pointer to the ``attach object stage`` from earlier. This allows the stage to know how the object is attached. We then create a ``ComputeIK`` stage and pass it our ``GeneratePlacePose`` stage - the rest follows the same logic as above with the pick stages.
```
Comment (henningkayser) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:760]:
```suggestion
This next stage generates the poses used to place the object and compute the inverse kinematics for those poses - it is somewhat similar to the ``generate grasp pose`` stage from the pick serial container. We start by creating a stage to generate the poses and inheriting the task properties. We specify the pose where we want to place the object with a ``PoseStamped`` message from ``geometry_msgs`` - in this case, we choose ``y = 0.5``. We then pass the target pose to the stage with ``setPose``.  Next, we use ``setMonitoredStage`` and pass it the pointer to the ``attach object stage`` from earlier. This allows the stage to know how the object is attached. We then create a ``ComputeIK`` stage and pass it our ``GeneratePlacePose`` stage - the rest follows the same logic as above with the pick stages.
```

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 11]
```diff
@@ -0,0 +1,883 @@
+Pick and Place with MoveIt Task Constructor
+===========================================
+
+.. image:: mtc.png
```
Comment (henningkayser) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:4]: I think it would be nice if this image was a gif

**Review by henningkayser — COMMENTED** [review]
This is great, I really like how it starts with the minimal example. This should really help people to get up-to-speed with some simple examples. I left some comments about possible improvements.  Also, the tutorial should probably reference the [concepts](https://moveit.picknik.ai/galactic/doc/examples/moveit_task_constructor/moveit_task_constructor_tutorial.html#basic-concepts) directly, without those it might be very hard to follow. That part in the link should probably be a concepts page as well, but this doesn't have to be done with this PR

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst (hunk 5 above)
Comment (stephanie-eng) [reply doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:462]: The indentation was wrong, which makes this look pretty strange (hopefully fixed now), but I think that brace is needed (matches with line 410)

**Comment by stephanie-eng** [comment]
For what it's worth, to prevent needing to scroll in the code blocks, I set my clang format line length to be shorter. Ideally, I think there's some value in having (mostly) line-for-line consistency between the code blocks and the source code, but that seems to make the clang-format test unhappy. Any thoughts on this?

**Comment by AndyZe** [comment]
> Any thoughts on this?

When manual formatting is preferred, you can disable clang for blocks of code (assuming you're talking about a cpp file).  https://stackoverflow.com/questions/33258159/how-to-make-clang-format-skip-sections-of-c-code

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/launch/mtc_demo.launch.py [hunk 12]
```diff
@@ -0,0 +1,110 @@
+import os
+from launch import LaunchDescription
+from launch.actions import ExecuteProcess
+from launch_ros.actions import Node
+from ament_index_python.packages import get_package_share_directory
+from moveit_configs_utils import MoveItConfigsBuilder
+
+
+def generate_launch_description():
```
Comment (AndyZe) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/launch/mtc_demo.launch.py:9]: Is this a direct copy of `panda_moveit_config demo.launch.py`? If so, let's delete it and just use that one.

If it's just a bit different, then you could call that launch file from this one. 

https://answers.ros.org/question/363687/ros2-call-launch-file-from-another-launch-file/

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 13]
```diff
@@ -0,0 +1,899 @@
...
+      geometry_msgs::msg::Pose pose;
+      pose.position.x = 0.5;
+      pose.position.y = -0.25;
+      object.pose = pose;
+
+      moveit::planning_interface::PlanningSceneInterface psi;
+      psi.applyCollisionObject(object);
+    }
+
+This function interfaces with the MoveIt Task Constructor task object. It first creates a task, which includes setting some properties and adding stages. This will be discussed further in the ``createTask`` function definition. Next, ``task.init()`` initializes the task and ``task.plan(5)`` generates a plan, stopping after 5 successful plans are found. The next line publishes the solution to be visualized in RViz - this line can be removed if you don't care for visualization. Finally, ``task.execute()`` executes the plan.
```
Comment (AndyZe) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:370]:
```suggestion
This function interfaces with the MoveIt Task Constructor task object. It first creates a task, which includes setting some properties and adding stages. This will be discussed further in the ``createTask`` function definition. Next, ``task.init()`` initializes the task and ``task.plan(5)`` generates a plan, stopping after 5 successful plans are found. The next line publishes the solution to be visualized in RViz - this line can be removed if you don't care for visualization. Finally, ``task.execute()`` executes the plan. Execution occurs via an action server interface with the RViz plugin.
```

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 14]
```diff
@@ -0,0 +1,899 @@
...
+      mtc::Stage* current_state_ptr = nullptr;  // Forward current_state on to grasp pose generator
+      auto stage_state_current = std::make_unique<mtc::stages::CurrentState>("current");
+      current_state_ptr = stage_state_current.get();
+      task.add(std::move(stage_state_current));
+
+In order to plan any robot motions, we need to specify a solver. MoveIt Task Constructor has three options for solvers:
+
+ * ``PipelinePlanner`` uses MoveIt's planning pipeline, which typically defaults to OMPL.
+ * ``CartesianPath`` is used to move the end effector in a straight line in Cartesian space.
+ * ``JointInterpolation`` is a simple planner that interpolates between the start and goal joint states. It is typically used for simple motions as it computes pretty fast, but it doesn't support complex motions.
```
Comment (AndyZe) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:437]:
```suggestion
 * ``JointInterpolation`` is a simple planner that interpolates between the start and goal joint states. It is typically used for simple motions as it computes fast but doesn't support complex motions.
```

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst [hunk 15]
```diff
@@ -0,0 +1,899 @@
...
+
+
+Congratulations! You've now defined a pick and place task using MoveIt Task Constructor!
+
+Visualizing with RViz
+---------------------
+
+The task with each comprising stage is shown in the Motion Planning Tasks pane. Click on a stage and additional information about the stage will show up to the right. The right pane shows different solutions as well as their associated costs. Depending on the stage type and the robot configuration, there may only be one solution shown.
+
+Click a solution to see an animation of the robot following the plan for that stage. Click the "Exec" button in the upper-right portion of the pane to execute the motion.
```
Comment (AndyZe) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst:891]: I've been confused by this in the past --- you need to click on the cost of a particular solution, then you can click Execute.

```suggestion
Click one of the solution costs to see an animation of the robot following the plan for that stage. Click the "Exec" button in the upper-right portion of the pane to execute the motion.
```

**Review by AndyZe — COMMENTED** [review]
Overall I think this is great. It's been useful to me already. If you make these small changes, I'll approve it.

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/launch/mtc_demo.launch.py (hunk 12 above)
Comment (stephanie-eng) [reply doc/tutorials/pick_and_place_with_moveit_task_constructor/launch/mtc_demo.launch.py:9]: This is an altered version of [moveit_task_constructor_demo demo.launch.py](https://github.com/ros-planning/moveit_task_constructor/blob/ros2/demo/launch/demo.launch.py)

There are definitely some similarities between the two, but it seems the panda_moveit_config launchfile will launch RViz with its own RViz custom config, so if I were to call that launch file from this one, I don't think I can launch RViz with the mtc.rviz config in this package.

Comment (AndyZe) [reply doc/tutorials/pick_and_place_with_moveit_task_constructor/launch/mtc_demo.launch.py:9]: ok, thanks for checking

**Review by AndyZe — APPROVED** [review]

**Comment by AndyZe** [comment]
CI is complaining, at least in part, b/c `panda_controllers.yaml` has been renamed to `panda_ros2_controllers.yaml`

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/src/main.cpp [hunk 16]
```diff
@@ -0,0 +1,352 @@
...
+      // Sample grasp pose
+      auto stage = std::make_unique<mtc::stages::GenerateGraspPose>("generate grasp pose");
+      stage->properties().configureInitFrom(mtc::Stage::PARENT);
+      stage->properties().set("marker_ns", "grasp_pose");
+      stage->setPreGraspPose("open");
+      stage->setObject("object");
+      stage->setAngleDelta(M_PI / 12);
+      stage->setMonitoredStage(current_state_ptr);  // Hook into current state
+
+      Eigen::Isometry3d grasp_frame_transform;
```
Comment (AndyZe) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/src/main.cpp:176]:
```suggestion
      // This is the transform from the object frame to the end-effector frame
      Eigen::Isometry3d grasp_frame_transform;
```

I was confused about this. Might as well add a comment

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/src/main.cpp [hunk 17]
```diff
@@ -0,0 +1,353 @@
...
+      "move to pick",
+      mtc::stages::Connect::GroupPlannerVector{ { arm_group_name, sampling_planner } });
+  stage_move_to_pick->setTimeout(5.0);
+  stage_move_to_pick->properties().configureInitFrom(mtc::Stage::PARENT);
+  task.add(std::move(stage_move_to_pick));
+
+  mtc::Stage* attach_object_stage =
+      nullptr;  // Forward attach_object_stage to place pose generator
+  {
+    auto grasp = std::make_unique<mtc::SerialContainer>("pick object");
```
Comment (AndyZe) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/src/main.cpp:142]: Note that I have just learned, all of these Stages can be inside the SerialContainer. I don't see any reason not to do that. I think it's confusing to have half the stages inside the container, half outside.

Suggest moving all stages into it.

Maybe @henningkayser or @JafarAbdi have a suggestion.

File: doc/tutorials/pick_and_place_with_moveit_task_constructor/src/main.cpp [hunk 18]
```diff
@@ -0,0 +1,353 @@
...
+  auto stage_move_to_pick = std::make_unique<mtc::stages::Connect>(
+      "move to pick",
+      mtc::stages::Connect::GroupPlannerVector{ { arm_group_name, sampling_planner } });
+  stage_move_to_pick->setTimeout(5.0);
+  stage_move_to_pick->properties().configureInitFrom(mtc::Stage::PARENT);
+  task.add(std::move(stage_move_to_pick));
+
+  mtc::Stage* attach_object_stage =
+      nullptr;  // Forward attach_object_stage to place pose generator
+  {
```
Comment (AndyZe) [inline doc/tutorials/pick_and_place_with_moveit_task_constructor/src/main.cpp:141]:
```suggestion

    // This is an example of SerialContainer usage. It's not strictly needed here.
    // In fact, `task` itself is a SerialContainer by default.
    {
```

I learned more about containers, per convo [here](https://github.com/ros-planning/moveit_task_constructor/issues/342). Might as well add this knowledge to the tutorial.

**Comment by stephanie-eng** [comment]
Related: https://github.com/ros-planning/moveit_task_constructor/pull/352
Required to pass CI.
